espnet/asr/pytorch_backend/asr_init.py

```diff
@@ -0,0 +1,234 @@
+import logging
+import os
+import torch
+
+from collections import OrderedDict
+
+from espnet.asr.asr_utils import get_model_conf
+from espnet.asr.asr_utils import torch_load
+
+from espnet.nets.asr_interface import ASRInterface
+from espnet.nets.mt_interface import MTInterface
+
+from espnet.utils.dynamic_import import dynamic_import
+
+
+def transfer_verification(model_state_dict, partial_state_dict, modules):
+    """Verify tuples (key, shape) for input model modules match specified modules.
+
+    Args:
+        model_state_dict (odict): the initial model state_dict
+        partial_state_dict (odict): the trained model state_dict
+        modules (list): specified module list for transfer
+
+    Return:
+        (boolean): allow transfer
+    """
+
+    modules_model = []
+    partial_modules = []
+
+    for key_p, value_p in partial_state_dict.items():
+        if any(key_p.startswith(m) for m in modules):
+            partial_modules += [(key_p, value_p.shape)]
+
+    for key_m, value_m in model_state_dict.items():
+        if any(key_m.startswith(m) for m in modules):
+            modules_model += [(key_m, value_m.shape)]
+
+    len_match = (len(modules_model) == len(partial_modules))
+    module_match = (sorted([x for x in modules_model]) ==
+                    sorted([x for x in partial_modules]))
+
+    return len_match and module_match
```

Comment:
You seem to skip shape matching. Is it OK? I think I added such parts...

Reply:
I don't skip shape matching, `modules_model` (main) and `partial_modules` (trained)  contains (name, shape) for matched in `modules`. Thus this test will return false if there is a shape mismatch!